src/drivers/ports/esp32/can_port.i

```diff
@@ -182,6 +185,13 @@ static void isr(void *arg_p)
         } else {
             thrd_resume_isr(self_p->thrd_p, 0);
             self_p->thrd_p = NULL;
+
+            higher_prio_task_woken = pdFALSE;
+            xSemaphoreGiveFromISR(thrd_idle_sem, &higher_prio_task_woken);
+
+            if (higher_prio_task_woken == pdTRUE) {
+                portYIELD_FROM_ISR() ;
+            }
```

Comment:
We should give the semaphore when data is received as well. I suggest that we unconditionally (or based on the interrupt result) give the semaphore at the end of the interrupt to start with. Incoming frames are written to a queue, and unfortunately the ``queue_write_isr``-function does not give any information about if the thread was resumed or not.